src/read.c

```diff
@@ -1739,7 +1739,10 @@ static avifBool avifParseColourInformationBox(avifProperty * prop, const uint8_t
     CHECK(avifROStreamRead(&s, colorType, 4));
     if (!memcmp(colorType, "rICC", 4) || !memcmp(colorType, "prof", 4)) {
         colr->hasICC = AVIF_TRUE;
-        colr->icc = avifROStreamCurrent(&s);
+        // Remember the offset of the ICC chunk relative to the beginning of the stream. A direct pointer cannot be stored because
+        // decoder->io->persistent could have been AVIF_FALSE when obtaining raw through decoder->io->read().
+        // The bytes could be copied now instead of remembering the offset, but it is as invasive as passing rawOffset everywhere.
+        colr->iccOffset = rawOffset + avifROStreamOffset(&s);
```

Comment:
at line 1939 rawOffset is passed in as `rawOffset + avifROStreamOffset(&s)` is this value correct?

Reply:
Yes, `rawOffset` is the offset at which you can find `raw` within the input bitstream. Should it have a different name? I am not so sure I follow what is unclear here.

Reply:
> Yes, `rawOffset` is the offset at which you can find `raw` within the input bitstream. Should it have a different name? I am not so sure I follow what is unclear here.

I was just wondering about the call setting `rawOffset = rawOffset + avifROStreamOffset(&s)` and then adding ` avifROStreamOffset(&s)` again here

Reply:
James: The value `rawOffset + avifROStreamOffset(&s)` is correct. Note that each of these functions also passes `avifROStreamCurrent(&s)` as the `raw` argument to other functions, and `avifROStreamCurrent(&s)` is equal to the original value of `raw` + `avifROStreamOffset(&s)`.

Reply:
Thanks I wasn't paying attention to the use of `avifROStreamCurrent`.